src/components/TableBody.js

```diff
@@ -105,10 +105,22 @@ class TableBody extends React.Component {
   };
 
   handleRowClick = (row, data, event) => {
-    // don't trigger onRowClick if the event was actually a row selection
```

Comment:
We can fix the edge case by adding

```
    let isExpandableIconClick = false;

    // We need to know if our row click was actually on our expandable icon
    if (event.target.nodeName === 'path' && event.target.parentNode.id === 'expandable-button') isExpandableIconClick = true;
    else if (event.target.id === 'expandable-button') isExpandableIconClick = true;
```

to the top of the `handleRowClick` function and `&& !isExpandableIconClick` to the `selectableRowsOnClick` conditional (line 114).

Reply:
Then in `TableSelectCell`, we need to add `id="expandable-button"` to the `KeyboardArrowRight` icon.

Reply:
@gabrielliwerant - good catch on the edge case.
I implemented the change as you specified in commit 511dc7c.

However looking at the rest of the `handleRowClick` function, we specifically skip calling `onRowClick` if the selectable icon is clicked, so I think it would be consistent to do the same here with the expandable icon. It also feels cleaner, and so I've added a second commit with this change (65dbadd).
This may be considered a breaking change though (expandable icon no longer triggering onRowClick), so I'll leave the decision to you, and would be happy to revert commit 65dbadd if you prefer your original suggestion.

Reply:
I like your change here, but I agree it could break code that relied on the click event previously. How about we keep your code, but instead of skipping the functionality, we add a deprecation warning via console and add back the `isExpandableIconClick ` bits? Then in a future release, we can break the backwards compatibility.

Reply:
@gabrielliwerant - instead of adding the `isExpandableIconClick` variable, and associated pieces that would need to be removed later, I added a single conditional to check for the deprecated condition that issues the warning and calls onRowClick explicitly. This should make it simpler to remove in a future release.

I copied the deprecated format from the only other instance of the word 'deprecated' in MUIDataTable.js, though downgraded to a warning for this case.

Reply:
Yes, good call, that does make it easier to remove, thank you!

Haha, looks like your search didn't include uppercase! I was thinking we would take after the convention here https://github.com/gregnb/mui-datatables/blob/master/src/MUIDataTable.js#L266. I like using `error` because I think we want it to stand out as something devs should change in advance of removing the functionality.

Also, while you're at it, could you add the new options to the `expandable-rows` example, so that we have something to point people to?

Reply:
The line you linked to (L266) is the same one I copied from.
I don't feel that `error` would be the right level here, as developers would not have any way of removing it. i.e. if you have expandableRows turned on, and onRowClick defined, then the warning/error will appear every time someone expands a row. There is no way to avoid it, even if you don't depend on the `onRowClick` execution on the expand action.
An `error` would indicate that something needs to be changed, but that may not be true for most use cases, and in this case you wouldn't have any way to clear the error.

I've added `expandableRowsOnClick: true` to the `expandableRows` example and considered doing the same for `selectableRows` but decided that this wouldn't be the default preference for most users. Perhaps I should add `selectableRowsOnClick: false` to that example, so that the property is more visible?

Reply:
Your point is well taken. My thinking was that because the error only shows on the occasion of both `expandableRpws` and `onRowClick`, it wouldn't be a disruptive message to too many people, and I was thinking of erring on the side of a strong notice for any functionality people might be relying upon (since it's easier to ignore warnings).

I suppose the more robust way to deprecate would be to wrap the deprecated functionality with a new option, which people could then switch on to avoid the deprecated behavior, as well as the error. However, I don't think we need to be that robust here, and I don't have a strong enough feeling on the matter to hold the PR for any more changes here.

Also, my thoughts on adding the options to the examples was just for the sake of demonstration rather than something representing real-world use cases, but I'm fine with the single addition for now.